core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -364,10 +370,13 @@ public ScheduledExecutorService scheduler() {
      *
      * @param command the action to take
      * @param interval the delay interval
-     * @return a ScheduledFuture who's get will return when the task is complete and throw an exception if it is canceled
+     * @return a ScheduledFuture who's get will block until the task has been canceled and then once canceled will throw a
```

Comment:
can we add a word about the implications of using SAME? i.e. this will run on the schedule thread, of which there is only one and should only do super light weight things.